test/unit/BPool/BPool_JoinswapPoolAmountOut.t.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {BPoolBase} from './BPoolBase.sol';
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+import {BNum} from 'contracts/BNum.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+
+contract BPoolJoinswapPoolAmountOut is BPoolBase, BNum {
+  address public tokenIn;
+
+  // Valid scenario
+  uint256 public poolAmountOut = 1e18;
+  uint256 public tokenInWeight = 8e18;
+  uint256 public totalWeight = 10e18;
+  uint256 public tokenInBalance = 300e18;
+  // ((((INIT_POOL_SUPPLY+poolAmountOut)/INIT_POOL_SUPPLY)^(1/(tokenInWeight/totalWeight)))*tokenInBalance-tokenInBalance)/(1-((1-(tokenInWeight/totalWeight))*MIN_FEE))
+  // ((((100+1)/100)^(1/(8/10)))*300-300)/(1-((1-(8/10))*(10^-6)))
+  // 3.754676583174615979425132956656691
+  uint256 public maxTokenIn = 3.754676583181324836e18;
```

Comment:
yes, they are different :upside_down_face: I prefer the difference between `BMath` and external calc to be explicit (knowing we're 10^-9 off) to be explicit like this to calling `BMath` directly